rotors_gazebo_plugins/src/gazebo_gps_plugin.cpp

```diff
@@ -190,9 +197,47 @@ void GazeboGpsPlugin::OnUpdate() {
 #else
   current_time = parent_sensor_->GetLastMeasurementTime();
 
-  gz_gps_message_.set_latitude(parent_sensor_->GetLatitude().Degree());
-  gz_gps_message_.set_longitude(parent_sensor_->GetLongitude().Degree());
-  gz_gps_message_.set_altitude(parent_sensor_->GetAltitude());
+  // Get the world pose of the sensor.
+  std::string parentName = parent_sensor_->GetParentName();
+  physics::EntityPtr parent = world_->GetEntity(parentName);
+  math::Pose sensorPose = parent_sensor_->GetPose() + parent->GetWorldPose();
```

Comment:
It think so.. when I convert the produced GPS coordinates back to the local coordinate system, it yields the same as the ground truth messages.
But yeah there might be a prettier way to do this.. but not sure how.